src/toast/Toast.tsx

```diff
@@ -0,0 +1,153 @@
+    SUCCESS = "success",
+    WARNING = "warning",
+    DANGER = "danger",
+    INFO = "info",
+}
+
+export enum Direction {
+    TOP_RIGHT = "top-right",
+    BOTTOM_LEFT = "bottom-left",
+    BOTTOM_RIGHT = "bottom-right",
+    TOP_LEFT = "top-left",
+    LEFT = "left",
+    RIGHT = "right",
+    BOTTOM = "bottom",
+    TOP = "top",
+    CAPS_LEFT = "Left",
+    CAPS_RIGHT = "Right",
+    CAPS_BOTTOM = "Bottom",
+    CAPS_TOP = "Top",
+}
+
+const MARGINS = {
+    POS_MARGIN: "15px",
+    NEG_MARGIN: "-15px",
+};
+
+const ICONTYPE = {
+    WARNING: "dell-alert-warning",
+    SUCCESS: "success-standard",
+    DANGER: "error-standard",
+    INFO: "dell-alert-info",
+};
+
+interface ToastProps {
+    notify: (fn: Function) => void;
+}
+export default class Toast extends React.Component<ToastProps> {
+    constructor(props: ToastProps) {
+        super(props);
+        props.displayToast(this.show_toast);
+
+        this.state = {
+            isToastRendering: false,
+            msg: "",
+            type: MessageType.SUCCESS,
+            position: Direction.TOP_RIGHT,
+            hideAfter: 3,
+            style: {opacity: 0},
+        };
+    }
+
+    /* Show the toaster unless another toaster is already in display */
+    show_toast = (message: string, options: object) => {
+        //Default display time for toast is 3 sec
+        //Default type is SUCCESS
+        //Default position is top-right
+        const hideTime = (options.hideAfter === undefined ? 3 : options.hideAfter) * 1000;
+        const toastType = options.type === undefined ? MessageType.SUCCESS : options.type;
+        const pos = options.position === undefined ? Direction.TOP_RIGHT : options.position;
+
+        //As this is a simple toast, allowing one toast to be shown at a time
+        if (this.state.isToastRendering === false) {
+            //create styles based on the position given
+            let place = (pos || Direction.TOP_RIGHT).includes(Direction.BOTTOM)
+                ? Direction.CAPS_BOTTOM
+                : Direction.CAPS_TOP;
+            const marginTypeForTopOrBottom = `margin${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.LEFT) ? Direction.CAPS_LEFT : Direction.CAPS_RIGHT;
+            const marginTypeForLeftOrRight = `margin${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.LEFT) ? Direction.LEFT : Direction.RIGHT;
+            const isLeftOrRight = `${place}`;
+            place = (pos || Direction.TOP_RIGHT).includes(Direction.BOTTOM) ? Direction.BOTTOM : Direction.TOP;
+            const isBottomOrTop = `${place}`;
+
+            //set the state to show the toast and setTimeout to hide the toast
+            this.setState(
+                {
+                    isToastRendering: true,
+                    msg: message,
+                    type: toastType,
+                    position: pos,
+                    hideAfter: hideTime,
+                    style: {
```

Comment:
I am still not getting why we need to put style in component state ?

Reply:
Removed style.